### PR TITLE
[CI] Enable HIP and CUDA plugins in GitHub Actions builds

### PR DESCRIPTION
**Pull request by alexbatashev**
Build all supported targets as part of the main build step. The idea is to build compiler toolchain once and distribute binary artifacts across machines to run the required testing.

Also this patch adds option to disable some tests, that require real hardware. Those must be moved to intel/llvm-test-suite, as instructed by [contributing guidelines](https://github.com/intel/llvm/blob/sycl/CONTRIBUTING.md).

### CODE REVIEW

**Comment by alexbatashev** [comment]
@jchlanda @AerialMantis I see two tests failing despite this seems to be the default environment according to GSG. Have you seen similar failures before?

https://github.com/intel/llvm/runs/4428671041?check_suite_focus=true

**Comment by alexbatashev** [comment]
Instructions to reproduce locally `check-clang` fail locally:

```bash
sudo docker run -it ghcr.io/intel/llvm/ubuntu2004_build:latest /bin/bash
mkdir /ws
cd /ws
git clone https://github.com/intel/llvm.git --depth=1
cd llvm
mkdir build
cd build
CUDA_LIB_PATH="/usr/local/cuda/lib64/stubs" python ../buildbot/configure.py --ci-defaults --cuda --hip
ninja check-clang
```

**Comment by bader** [comment]
@alexbatashev, https://github.com/intel/llvm/pull/5103 should fix `check-clang`.

File: .github/workflows/sycl_linux_build_and_test.yml [hunk 1]
```diff
@@ -128,6 +131,8 @@ jobs:
     - name: check-sycl
       if: always()
       run: |
+        # TODO consider moving this to Dockerfile
+        export LD_LIBRARY_PATH=/usr/local/cuda/compat/:/usr/local/cuda/lib64:$LD_LIBRARY_PATH
```
Comment (bader) [inline .github/workflows/sycl_linux_build_and_test.yml:135]: According to https://intel.github.io/llvm-docs/GetStartedGuide.html#build-dpc-toolchain-with-support-for-nvidia-cuda, this should not be required. I suggest filing an issue to investigate what is wrong here. Maybe NVIDIA docker image is not configured correctly.

I have the same question regarding `--hip-amd-arch` option. I don't see it listed as required option in the GetStartedGuide. Should we add it there?

**Review by bader — APPROVED** [review]

File: .github/workflows/sycl_linux_build_and_test.yml (hunk 1 above)
Comment (alexbatashev) [reply .github/workflows/sycl_linux_build_and_test.yml:135]: > Maybe NVIDIA docker image is not configured correctly.

That is actually the case

> I have the same question regarding --hip-amd-arch option. I don't see it listed as required option in the GetStartedGuide. Should we add it there?

It is mentioned here: https://intel.github.io/llvm-docs/GetStartedGuide.html#run-in-tree-lit-tests
The option is not required to build the compiler, only to test it.

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**